### PR TITLE
change the use of environment variables

### PR DESCRIPTION
**Pull request by sinisaos**
@dantownsend What do you think about the approach of using environment variables in this PR. Another option is to not use the `.env` file but pass the env vars to docker run like this 
```bash
docker run -d -e DB_NAME='some_db_name' \
 -e DB_PASSWORD='some_db_password' \
 -e DB_USER='some_db_user' \
 -e DB_HOST='some_db_host' \
 -e DB_PORT='some_db_port' \
 -e USERNAME='piccolo' \
 -e PASSWORD='piccolo123' \
 -e EMAIL='piccolo@piccolo.com' \
 --network=host --name admin_container piccolo_admin 
```
I'm not a Docker expert, but I don't find it convenient to use `ENV` vars in `Dockerfile`. In docs is also a warning `"Note that, because the environment variables you set persist in containers, using them can lead to unintended side-effects for the application's runtime."`

### CODE REVIEW

**Comment by dantownsend** [comment]
@sinisaos I think we need to allow the users to pass in environment variables. Even though it's a pain to pass them in via the command line, most users with have something like Docker Compose, or Kubernetes, where the config is a YAML file, which makes it really easy to specify environment variables.

There will reach a point though where environment variables alone becomes unmanageable. Piccolo Admin has so many potential config options, that we will need to allow some kind of config file (YAML or something). We could let the user add the config file to the container, or have some kind of UI which guides the user through the options, and generates the config. Or maybe even stores the config within the SQLite database.

I have this repo from a few years ago where I experimented with this a bit:

https://github.com/piccolo-orm/piccolo_admin_docker

The repo is a mess though. I started playing around with a config UI:

https://github.com/piccolo-orm/piccolo_admin_docker/tree/main/config_tool

But didn't get very far. I was using a library which generated UI from a JSON schema. I thought it would save time, but honestly it is kind of a pain. So if we built a config tool, we may as well just build the UI ourselves from scratch.

**Comment by dantownsend** [comment]
@sinisaos I just gave you admin permissions on this repo - sorry, I didn't realise you must have lost them when transferring it to the Piccolo org.

**Comment by sinisaos** [comment]
@dantownsend I agree some sort of UI would be great. Unfortunately the links you post are returning a `404`. I find using an `.env` file to be most convenient, because the user just needs to fill in that `.env` file and run `cp .env.example .env && rm .env.example` and they ready to go. Docker compose would be almost the same with something like this
```yaml
services:
  web:
    build: .
    command: python app/main.py
    volumes:
      - .:/app
    env_file:
      - .env
```
Docker `run` also has an interactive mode flag (`-it`) with which we can ask for user input similar to piccolo asgi new. I'm experimenting with it a bit so I wanted to hear your thoughts. Thanks for the answer.

**Comment by sinisaos** [comment]
@dantownsend sorry to interrupt, but what do you think about the latest changes. Is this a good direction to use Piccolo Admin config options (using a yaml file)?

**Comment by dantownsend** [comment]
@sinisaos I think having a YAML config file makes sense - having everything as an environment variable would be unmanageable.

The closest comparable thing I have worked with is self hosted GitLab, which all runs in a Docker container. They use a mix of environment variables and a Ruby config file which is mounted inside.

**Comment by sinisaos** [comment]
> The closest comparable thing I have worked with is self hosted GitLab, which all runs in a Docker container. They use a mix of environment variables and a Ruby config file which is mounted inside.

@dantownsend I think we have a similar thing here. Secrets (sensitive information) are stored in an `.env` file which I think is a good practice. User configuration (how user wants to display columns and other things) is stored in `yaml` file which is added to container. How the user should format the yaml file is shown in the readme as an example. The user must read the Piccolo Admin docs to see how it works, but basically with these changes we can use Piccolo Admin with an existing (legacy) database if Piccolo Admin supports database columns. I will also add `rich_text_columns` (user needs to read docs to see only `Text` column is supported to get `WYSIWYG` editor) and `link_column` (again user needs to read docs to see that if `link_column` is enabled, `primary_key` column should be disabled (not used)).

**Comment by sinisaos** [comment]
@dantownsend I will merge this PR. Everything from this PR works on localhost (I don't have the ability to test with online database, but I think it should work if `network_mode: "host"` from `docker-compose.yml` is removed), but feel free to make changes as you think is best.

File: app/main.py [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
 from piccolo_api.session_auth.tables import SessionsBase
 
-DB = SQLiteEngine()
-
+with open("config.yaml") as stream:
+    try:
+        admin_config = yaml.safe_load(stream)
+        BASE_CONFIG = admin_config.get("tables")
+    except yaml.YAMLError as exc:
+        raise exc
```
Comment (dantownsend) [inline app/main.py:25]: I know you've merged this in, but in a future PR you could define the schema as a Pydantic model, and then pass the loaded YAML into that.

File: README.md [hunk 2]
```diff
@@ -11,20 +11,77 @@ git clone https://github.com/piccolo-orm/piccolo-admin-docker.git
...
+
+After Piccolo Admin is started with all the tables from the existing database, we can do additional configuration through the `config.yaml` file. 
+
+Example of `config.yaml`:
+
+```yaml
+tables:
+  # An example of additional Piccolo Admin configuration
+  Actor:
+    visible_columns:
```
Comment (dantownsend) [inline README.md:43]: Maybe tweak this, so instead tables can either just be a string (the name of the table), or an object. For example:

``` yaml
-  city
-  name: address
    visible_columns:
      - address_id
      - address
      - city_id
    visible_filters:
      - address_id
      - address
      - city_id
    menu_group: Location
    rich_text_columns: address
```

So a user doesn't have to specify all of the options if they don't want to.

File: app/main.py (hunk 1 above)
Comment (sinisaos) [reply app/main.py:25]: You are right. I hadn't thought about Pydantic at all. With the Pydantic model parsing yaml will be much cleaner than it is now with dict. I will try to do it tomorrow. Thanks.

**Pull request merged**